### PR TITLE
Update vita toolchain

### PR DESCRIPTION
**Pull request by crudelios**
Update vita toolchain, which uses a more recent/better integrated SDL build.

Allows getting rid of the `vita2d` hacks and improves performace on Vita by 5% to 10%.

### CODE REVIEW

File: src/platform/screen.h [hunk 1]
```diff
@@ -18,7 +18,8 @@ void platform_screen_center_window(void);
 void platform_screen_recreate_texture(void);
 #endif
 
-void platform_screen_render(void);
+void platform_screen_render(int update_screen);
```
Comment (bvschaik) [inline src/platform/screen.h:21]: Can we please eliminate this boolean flag by creating two functions?
Perhaps `platform_screen_draw_and_render(void)` (original code) and `platform_screen_render(void)` with just the `SDL_RenderPresent()`.

For rationale, see Clean Code by Robert C. Martin, or [this article](https://martinfowler.com/bliki/FlagArgument.html) by Martin Fowler.

File: src/platform/vita/vita_keyboard.c [hunk 2]
```diff
@@ -109,27 +110,20 @@ const uint8_t *vita_keyboard_get(const uint8_t *initial_text, int max_length)
...
         if (ime_result == IME_DIALOG_RESULT_FINISHED) {
             encoding_from_utf8(ime_text_utf8, final_text, max_length);
         } else if (ime_result != IME_DIALOG_RESULT_CANCELED) {
             done = 0;
         }
-        vita2d_end_drawing();
-        vita2d_common_dialog_update();
-        vita2d_swap_buffers();
-        sceDisplayWaitVblankStart();
+        platform_screen_render(0);
```
Comment (bvschaik) [inline src/platform/vita/vita_keyboard.c:126]: Just wondering: what's the reason for not wanting to redraw the screen at this point?

**Review by bvschaik — CHANGES_REQUESTED** [review]
Generally looking good. Nice to be rid of the vita-specific code.
One requested improvement and one question.

File: src/platform/screen.h (hunk 1 above)
Comment (crudelios) [reply src/platform/screen.h:21]: I completely agree. I was thinking about doing just that when I had some time 😉 

I was thinking about `platform_screen_update` and `platform_screen_render`. Normal code would call both functions, the vita keyboard code would just need to call `render`.

File: src/platform/vita/vita_keyboard.c (hunk 2 above)
Comment (crudelios) [reply src/platform/vita/vita_keyboard.c:126]: The normal redraw gets the drawing from our texture which is controlled by `graphics_canvas`. When drawing the virtual keyboard, we don't want/need the texture from there but rather the internal rendering from vita's virtual keyboard.  So we only need to call renderer but not to update the texture.

File: src/platform/screen.h (hunk 1 above)
Comment (bvschaik) [reply src/platform/screen.h:21]: Sounds good 👍

**Pull request merged**